hyperspy/io_plugins/protochips.py

```diff
@@ -0,0 +1,183 @@
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with HyperSpy.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import numpy as np
+import os
+from datetime import datetime
+import warnings
+
+# Plugin characteristics
+# ----------------------
+format_name = 'Protochips'
+description = 'Reads Protochips log files (heating/baising and gas cell)'
+full_support = False
+# Recognised file extension
+file_extensions = ['csv', 'CSV']
+default_extension = 0
+# Reading capabilities
+reads_images = False
+reads_spectrum = False
+reads_spectrum_image = False
+# Writing capabilities
+writes = False
+
+
+def file_reader(filename, *args, **kwds):
+    csv_file = ProtochipsCSV(filename)
+    return protochips_log_reader(csv_file)
+
+
+def protochips_log_reader(csv_file):
+    csvs = []
+    for key in csv_file.logged_quantity_name_list:
+        csvs.append(csv_file.get_dictionary(key))
+    return csvs
+
+
+class ProtochipsCSV(object):
+
+    def __init__(self, filename, header_line_number=10):
+        self.filename = filename
+        self.raw_header = self._read_header(header_line_number)
+        self.column_name = self._read_column_name()
+        self._read_all_metadata_header()
+        self.logged_quantity_name_list = self.column_name[2:]
+
+        self._read_data(header_line_number)
+
+        self.calibration_file = None
+
+    def get_dictionary(self, quantity):
+        return {'data': self._data_dictionary[quantity],
+                'axes': self._get_axes(),
+                'metadata': self._get_metadata(quantity),
+                'original_metadata': {'Protochips_header':
+                                      self._get_original_metadata()},
+                'mapping': self._get_mapping()}
+
+    def _get_original_metadata(self):
+        d = {'Start time': self.start_datetime}
+        d['Time units'] = self.time_units
+        for quantity in self.logged_quantity_name_list:
+            d['%s_units' % quantity] = self._get_quantity_units(quantity)
+        d['User'] = self.user
+        d['Calibration file name'] = self._parse_calibration_file()
+        return d
+
+    def _get_mapping(self):
+        return {'Protochips_header.date': ("General.time", None), }
+
+    def _get_metadata(self, quantity):
+        return {'General': {'original_filename': os.path.split(self.filename)[1],
+                            'title': '%s (%s)' % (quantity,
+                                                  self._get_quantity_units(quantity)),
+                            'user': self.user,
+                            'start_time': self.start_datetime,
+                            'notes': self._parse_notes(),
+                            'calibration_file': self._parse_calibration_file()},
+                "Signal": {'signal_type': quantity,
```

Comment:
Should be signal_type an empty string? Or this is alright to set a physical quantity, such as temperature, current, etc, in case it can be use in the future?